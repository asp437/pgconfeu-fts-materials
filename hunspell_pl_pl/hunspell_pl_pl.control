# hunspell_pl_pl extension
comment = 'Polish Hunspell Dictionary'
default_version = '1.0'
module_pathname = '$libdir/hunspell_pl_pl'
relocatable = true
