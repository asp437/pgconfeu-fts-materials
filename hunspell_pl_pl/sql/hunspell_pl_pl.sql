CREATE EXTENSION hunspell_pl_pl;

CREATE TABLE table1(name varchar);
INSERT INTO table1 VALUES ('ziemia'), ('ziemie'),
						('liści'), ('lista'), ('biegać'), ('uciec'),
						('bieganie');

SELECT d.* FROM table1 AS t, LATERAL ts_debug('polish_hunspell', t.name) AS d;

CREATE INDEX name_idx ON table1 USING GIN (to_tsvector('polish_hunspell', "name"));
SELECT * FROM table1 WHERE to_tsvector('polish_hunspell', name)
	@@ to_tsquery('polish_hunspell', 'ziemia');
SELECT * FROM table1 WHERE to_tsvector('polish_hunspell', name)
	@@ to_tsquery('polish_hunspell', 'liści');
SELECT * FROM table1 WHERE to_tsvector('polish_hunspell', name)
	@@ to_tsquery('polish_hunspell', 'bieganie');

DROP INDEX name_idx;
CREATE INDEX name_idx ON table1 USING GIST (to_tsvector('polish_hunspell', "name"));
SELECT * FROM table1 WHERE to_tsvector('polish_hunspell', name)
	@@ to_tsquery('polish_hunspell', 'ziemia');
SELECT * FROM table1 WHERE to_tsvector('polish_hunspell', name)
	@@ to_tsquery('polish_hunspell', 'liści');
SELECT * FROM table1 WHERE to_tsvector('polish_hunspell', name)
	@@ to_tsquery('polish_hunspell', 'bieganie');
