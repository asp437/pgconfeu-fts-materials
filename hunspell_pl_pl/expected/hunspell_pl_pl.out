CREATE EXTENSION hunspell_ru_ru;
CREATE TABLE table1(name varchar);
INSERT INTO table1 VALUES ('земля'), ('землей'), ('землями'), ('земли'),
						('туши'), ('тушь'), ('туша'), ('тушат'),
						('тушью');
SELECT d.* FROM table1 AS t, LATERAL ts_debug('russian_hunspell', t.name) AS d;
 alias |    description    |  token  |          dictionaries           |    dictionary    |        lexemes         
-------+-------------------+---------+---------------------------------+------------------+------------------------
 word  | Word, all letters | земля   | {russian_hunspell,russian_stem} | russian_hunspell | {земля}
 word  | Word, all letters | землей  | {russian_hunspell,russian_stem} | russian_hunspell | {земля}
 word  | Word, all letters | землями | {russian_hunspell,russian_stem} | russian_hunspell | {земля}
 word  | Word, all letters | земли   | {russian_hunspell,russian_stem} | russian_hunspell | {земля}
 word  | Word, all letters | туши    | {russian_hunspell,russian_stem} | russian_hunspell | {туша,тушь,тушить,туш}
 word  | Word, all letters | тушь    | {russian_hunspell,russian_stem} | russian_hunspell | {тушь}
 word  | Word, all letters | туша    | {russian_hunspell,russian_stem} | russian_hunspell | {туша,тушить,туш}
 word  | Word, all letters | тушат   | {russian_hunspell,russian_stem} | russian_hunspell | {тушить}
 word  | Word, all letters | тушью   | {russian_hunspell,russian_stem} | russian_hunspell | {тушь}
(9 rows)

CREATE INDEX name_idx ON table1 USING GIN (to_tsvector('russian_hunspell', "name"));
SELECT * FROM table1 WHERE to_tsvector('russian_hunspell', name)
	@@ to_tsquery('russian_hunspell', 'землей');
  name   
---------
 земля
 землей
 землями
 земли
(4 rows)

SELECT * FROM table1 WHERE to_tsvector('russian_hunspell', name)
	@@ to_tsquery('russian_hunspell', 'тушь');
 name  
-------
 туши
 тушь
 тушью
(3 rows)

SELECT * FROM table1 WHERE to_tsvector('russian_hunspell', name)
	@@ to_tsquery('russian_hunspell', 'туша');
 name  
-------
 туши
 туша
 тушат
(3 rows)

DROP INDEX name_idx;
CREATE INDEX name_idx ON table1 USING GIST (to_tsvector('russian_hunspell', "name"));
SELECT * FROM table1 WHERE to_tsvector('russian_hunspell', name)
	@@ to_tsquery('russian_hunspell', 'землей');
  name   
---------
 земля
 землей
 землями
 земли
(4 rows)

SELECT * FROM table1 WHERE to_tsvector('russian_hunspell', name)
	@@ to_tsquery('russian_hunspell', 'тушь');
 name  
-------
 туши
 тушь
 тушью
(3 rows)

SELECT * FROM table1 WHERE to_tsvector('russian_hunspell', name)
	@@ to_tsquery('russian_hunspell', 'туша');
 name  
-------
 туши
 туша
 тушат
(3 rows)

